pkgs/desktops/deepin/dde-file-manager/default.nix

```diff
@@ -0,0 +1,235 @@
+{ stdenv, fetchFromGitHub, fetchpatch, pkgconfig, qmake, qttools,
+  qtmultimedia, qtx11extras, gnome3, gsettings-qt, dtkcore, dtkwidget,
+  deepin-movie-reborn, jemalloc, ffmpegthumbnailer, poppler, taglib,
+  libsecret, kcodecs, deepin-gettext-tools, dde-polkit-agent, deepin,
+  qt5integration, gtk2, qtsvg, file, avfs, polkit, polkit-qt,
+  deepin-shortcut-viewer, xdg-user-dirs, dde-qt-dbus-factory,
+  deepin-terminal, gst-plugins-good, mpv, zlib, libX11, gvfs, glib,
+  xorg, udisks2-qt5, dde-daemon, dde-dock, deepin-anything,
+  autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "dde-file-manager";
+  version = "4.8.4";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
+    rev = version;
+    sha256 = "0yhy36gkk5nyn56sn1v3hjri22nxs9k7raflbszahm574xpr1akb";
+  };
+
+  nativeBuildInputs = [
+    deepin.setupHook
+    qmake                   # build
+    qttools                 # build
+    pkgconfig               # build
+    deepin-gettext-tools    # build
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    avfs                   # recommended
+    dde-dock                # build
+    dde-polkit-agent
+    dde-qt-dbus-factory    # build arch ?
+    deepin-anything        # build run [amd64 i386]
+    deepin-movie-reborn      # build arch
+    deepin-shortcut-viewer   # arch fedora ?
+    deepin-terminal          # arch fedora ?
+    dtkcore
+    dtkcore
+    dtkwidget             # build
+    ffmpegthumbnailer     # build
+    file                  # build arch fedora ?
+    #gnome3.file-roller    # arch fedora ?
+    gnome3.libsecret      # build
+    gsettings-qt          # build
+    #gst-plugins-good      # arch fedora ?
+    jemalloc              # build
+    kcodecs               # build arch ?
+    #mpv                   # arch ?
+    polkit                # build fedora
+    polkit-qt             # build
+    poppler               # build
+    qt5integration        # recommended
+    qtmultimedia          # build
+    qtsvg                 # build
+    qtx11extras           # build
+    taglib                # build fedora ?
+    udisks2-qt5           # build
+    zlib                  # run
+    #xdg-user-dirs         # arch fedora ?
+    ###!deepin-qt5config      # build
+
+    #libuchardet          # fedora ?
+    glib.dev              # build ?
+    glib.bin              # provides bin/gio, needed at runtime
+    xorg.libX11           # build ?
+    xorg.xorgproto        # build ?
+    xorg.libxcb           # build ?
+    xorg.xcbutilwm        # build ?
+    xorg.xcbutil          # build fedora ?
+
+    # run command by QProcess
+    #deepin-desktop       # fedora
+    gvfs # gvfs-client    # fedora ✓
+    #samba                # fedora
```

Comment:
Think this for sure can need samba, but probably not to link against.

Reply:
Keep samba. `/usr/sbin/smbd` is used.